arangod/Agency/FailedLeader.cpp

```diff
@@ -297,9 +302,15 @@ bool FailedLeader::start(bool& aborts) {
             if (s.size() > 0 && s[0] == '_') {
               s = s.substr(1);
             }
-            if (s != _from && s != _to) {
-              TRI_ASSERT(std::find(planv.begin(), planv.end(), s) != planv.end());
-              // A server in Current ought to be in the Plan, if not, we want to know this.
+            // We need to make sure to only pick servers from the plan as followers.
+            // There is a chance, that a server is removed from the plan, but it is not yet taken out
+            // from the in-sync followers in current.
+            // e.g. User Reduces the ReplicationFactor => Follower F1 will be taken from the Plan
+            // Now F1 drops the shard.
+            // For some Reason Leader cannot report out-of-sync, and dies.
+            // => F1 will be readded in shard, as an early follower, and is considered to be in sync, until
+            // New Leader has updated sync information.
+            if (s != _from && s != _to && std::find(planv.begin(), planv.end(), s) != planv.end()) {
```

Comment:
Just curious: are we losing test/debug information by removing the assert?

Reply:
No because the assert is unfortunately wrong. The situation asserted here is perfectly valid, and expected.
The timeframe where it can happen is a single maintenance roundtrip, so highly unlikely to be seen unless the server is under high maintenance load.